### DISCUSSION TITLE
Dynamic presets for adetailer.

### DISCUSSION THREAD

**Discussion by Symbiomatrix** [Bing-su/adetailer#278] [Ideas]
Currently, adetailer offers a constant number of generic sets of masking parameters in separate ui elements (in terms of id).
I propose the following design: Dynamic named and saved presets of mask parameters, each of which can be separately enabled / disabled at will, modified through a single set of ui elements. During processing, parameters will be read from any presets currently activated.
I think this could be an improvement for the following reasons:
- No copypasta of ui elements and boilerplate to support that (or less boilerplate).
- After detail prompts tend to be more consistent across generations and runs, at least in my workflow; if not the prompt, then 90% of the values besides it in the mask. It would be useful to have a "Face X" preset saved up and whatnot, to be enabled or edited at will.
- No limit on number of masks, and no need to reset ui to change number of masks.

I have yet to examine the code deeply, any thoughts would be appreciated.

**Comment by Bing-su** [comment]
adetailer의 기능을 벗어납니다. 다른 확장을 사용하세요.

**Reply by Symbiomatrix** [reply]
To clarify, would you object to a PR of such a feature?
